vyper/builtin_functions/functions.py

```diff
@@ -731,24 +744,32 @@ def evaluate(self, node):
         if " " in args[0].value:
             raise InvalidLiteral("Invalid function signature - no spaces allowed.")
 
+        return_type = self.infer_kwarg_types(node)
+        value = abi_method_id(args[0].value)
+
+        if isinstance(return_type, Bytes4Definition):
+            return vy_ast.Hex.from_node(node, value=hex(value))
+        else:
+            return vy_ast.Bytes.from_node(node, value=value.to_bytes(4, "big"))
+
+    def fetch_call_return(self, node):
+        validate_call_args(node, 1, ["output_type"])
+
+        type_ = self.infer_kwarg_types(node)
+        return type_
+
+    def infer_kwarg_types(self, node):
         if node.keywords:
             return_type = get_type_from_annotation(node.keywords[0].value, DataLocation.UNSET)
             if isinstance(return_type, Bytes4Definition):
-                is_bytes4 = True
+                return Bytes4Definition()
             elif isinstance(return_type, BytesArrayDefinition) and return_type.length == 4:
-                is_bytes4 = False
+                return BytesArrayDefinition(4)
             else:
                 raise ArgumentException("output_type must be Bytes[4] or bytes4", node.keywords[0])
         else:
             # If `output_type` is not given, default to `Bytes[4]`
-            is_bytes4 = False
-
-        value = abi_method_id(args[0].value)
-
-        if is_bytes4:
-            return vy_ast.Hex.from_node(node, value=hex(value))
-        else:
-            return vy_ast.Bytes.from_node(node, value=value.to_bytes(4, "big"))
+            return BytesArrayDefinition(4)
```

Comment:
nit: if this is the final branch and all the other branches fastpath out, don't use an explicit `else`. this makes it clearer what the default behavior is